transaction-status/src/option_serializer.rs

```diff
@@ -0,0 +1,89 @@
+use serde::{ser::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug, PartialEq)]
+pub struct OptionalValue<T> {
+    #[serde(
+        flatten,
+        default = "OptionSerializer::skip",
```

Comment:
@jstarry this will actually be a behavior change if we use OptionalValue for an older field, like `inner_instructions`.
Currently, if a client deserializes a transaction from before `inner_instructions` were added, the field will be deserialized as default, an empty Vec. Then if the client reserializes the struct, the field will be included as an empty array.
With your changes, `inner_instructions` will be deserialized as Skip, and if reserialized, the field will be skipped.
Do we care?

Reply:
> Currently, if a client deserializes a transaction from before inner_instructions were added, the field will be deserialized as default, an empty Vec

I'm not sure I follow the situation that you're describing. Is that because `inner_instructions` doesn't have the `#[serde(default)]` attribute?